packages/itwinui-css/src/button/button.scss

```diff
@@ -22,49 +19,99 @@ $iui-button-padding-large: $iui-xs * 6;
   position: relative;
   border-radius: $iui-border-radius;
   line-height: $iui-line-height;
-  box-shadow: none;
   font-size: $iui-font-size;
   font-weight: $iui-font-weight-normal;
-  text-decoration: none;
   user-select: none;
   cursor: pointer;
   white-space: nowrap;
-  border: 1px solid transparent;
+  border: 1px solid var(--_iui-button-border-color);
+  background-color: var(--_iui-button-background-color);
   color: var(--_iui-button-text-color);
-  @media (prefers-reduced-motion: no-preference) {
-    transition: color $iui-speed-fast ease-out, background-color $iui-speed-fast ease-out,
-      border-color $iui-speed-fast ease-out;
+  transition: background-color $iui-speed-fast ease-out, border-color $iui-speed-fast ease-out;
+  -webkit-tap-highlight-color: transparent;
+
+  @at-root a#{&} {
+    text-decoration: none;
   }
 
-  &:hover {
-    --_iui-button-text-color: rgba(var(--iui-color-foreground-body-rgb), var(--iui-opacity-1));
+  // #region Sizing
+  @include iui-button-size;
 
-    text-decoration: none;
+  &[data-iui-button-size='small'] {
+    @include iui-button-size(small);
   }
 
-  @include iui-focus;
+  &[data-iui-button-size='large'] {
+    @include iui-button-size(large);
+  }
+  // #endregion Sizing
+
+  &[aria-disabled='true'] {
+    @include iui-focus;
+  }
 
   &[disabled],
-  &:disabled {
-    @include iui-button-disabled;
+  &:disabled,
+  &[aria-disabled='true'] {
+    cursor: not-allowed;
+
+    &,
+    &:hover,
+    &:active {
+      --_iui-button-text-color: var(--iui-text-color-placeholder);
+      --_iui-button-background-color: var(--iui-color-background-disabled);
+      --_iui-button-border-color: var(--iui-color-background-disabled);
+    }
+  }
+}
+
+@mixin iui-button-label {
+  white-space: nowrap;
+  color: inherit;
+}
+
+@mixin iui-button-icon {
+  width: $iui-icons-default;
+  height: $iui-icons-default;
+  transition: fill $iui-speed-fast ease-out;
+  fill: currentColor;
+
+  @at-root [data-iui-button-variant='borderless'] &:only-child {
+    margin-left: $iui-component-offset;
+    margin-right: $iui-component-offset;
   }
 }
 
 @mixin iui-button-active {
+  &,
+  &:hover,
+  &:active {
+    --_iui-button-text-color: var(--iui-color-foreground-primary);
+  }
+
   // Shows active stripe only if --_iui-button-active-stripe-inset is set (from outside)
   &::after {
     content: '';
     position: absolute;
     inset: var(--_iui-button-active-stripe-inset);
-    background-color: var(--iui-color-foreground-primary);
+    background-color: currentColor;
   }
+}
 
-  &[disabled],
-  &:disabled {
-    &::after {
-      background-color: rgba(var(--iui-color-foreground-body-rgb), var(--iui-opacity-5));
+@mixin iui-button-dropdown {
+  padding-right: $iui-s;
+
+  &[data-iui-button-size='small'] {
+    padding-right: $iui-xs;
+
+    &[data-iui-button-variant='borderless'] {
+      padding-right: $iui-xxs;
```

Comment:
I've updated `variant.scss` to accept the button variant name as `@arg` instead of color.  Personally I think the older way was cleaner code wise but the `@arg`s make more sense this new way.